### PR TITLE
internal/blocktest: add package for shared test code

### PR DESCRIPTION
**Pull request by jsvisa**
dry

### CODE REVIEW

**Comment by jsvisa** [comment]
@rjl493456442  please take a look, thanks.

File: internal/blocktest/test_hash_test.go [hunk 1]
```diff
@@ -0,0 +1,47 @@
+// Copyright 2023 The go-ethereum Authors
```
Comment (holiman) [inline internal/blocktest/test_hash_test.go:1]: Please omit this file, no need to test this

File: internal/blocktest/test_hash.go [hunk 2]
```diff
@@ -0,0 +1,59 @@
...
+// You should have received a copy of the GNU Lesser General Public License
+// along with the go-ethereum library. If not, see <http://www.gnu.org/licenses/>.
+
+// Package utesting provides a standalone replacement for package testing.
+//
+// This package exists because package testing cannot easily be embedded into a
+// standalone go program. It provides an API that mirrors the standard library
+// testing API.
+
+package blocktest
```
Comment (fjl) [inline internal/blocktest/test_hash.go:23]: Since this package is only used for testing, there is no need to create a test here.

**Review by fjl — CHANGES_REQUESTED** [review]

**Review by holiman — APPROVED** [review]
LGTM after the `test_hash_test.go` is removed

File: internal/blocktest/test_hash_test.go (hunk 1 above)
Comment (jsvisa) [reply internal/blocktest/test_hash_test.go]: done

File: internal/blocktest/test_hash.go (hunk 2 above)
Comment (jsvisa) [reply internal/blocktest/test_hash.go:23]: sorry for that, omitted.

**Review by holiman — APPROVED** [review]

**Review by fjl — APPROVED** [review]

**Pull request merged**